clang/lib/Serialization/ASTWriter.cpp

```diff
@@ -4905,6 +4905,10 @@ ASTFileSignature ASTWriter::WriteAST(Sema &SemaRef, StringRef OutputFile,
   this->BaseDirectory.clear();
 
   WritingAST = false;
+
+  if (WritingModule)
+    updateModuleTimestamp(OutputFile);
```

Comment:
Calling this here is a bit annoying, since it will create timestamp files even for explicitly-built PCM files. I might move it somewhere into `CompilerInstance` where it specifically handle implicit builds. That will increase the delay between accessing the input files and writing the timestamp file. What do people think about that?

Reply:
Yeah, I'd like to make such changes doesn't affect C++20 modules if possible.

Reply:
Don't have a strong opinion about the location of updating the timestamp. 

To be consistent with another usage of `updateModuleTimestamp` shouldn't you check
```c++
HeaderSearchOptions &HSOpts = PP.getHeaderSearchInfo().getHeaderSearchOpts();
if (HSOpts.ModulesValidateOncePerBuildSession)
```

? And I hope it can help with the explicitly-built .pcm files.